api/agent/drivers/docker/docker.go

```diff
@@ -101,13 +148,36 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	imagesBeforeLoad, err := driver.docker.ListImages(context.Background())
+
 	if conf.DockerLoadFile != "" {
 		err = loadDockerImages(driver, conf.DockerLoadFile)
 		if err != nil {
 			logrus.WithError(err).Fatalf("cannot load docker images in %s", conf.DockerLoadFile)
 		}
 	}
 
+	if conf.MaxImageCacheSize != 0 {
+		driver.imageCache = NewCache()
+
+		go func(context context.Context) {
```

Comment:
I think there's a timing issue here. Since go-routine, this means, new images are being pulled as this is running... So, a user function can potentially be flagged as 'base image'